pkg/metainfo/kvmetainfo/metainfo.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package kvmetainfo
+
+import (
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storage"
+)
+
+var errClass = errs.Class("kvmetainfo")
+
+const defaultSegmentLimit = 8 // TODO
+
+var _ storj.Metainfo = (*DB)(nil)
+
+// DB implements metainfo database
+type DB struct {
+	*Buckets
+	*Objects
+}
+
+/*
+// New creates a new metainfo database
+func New(client psclient.Client, rootKey *storj.Key) *DB {
+	return &DB{
+		// Buckets: NewBuckets(buckets),
+		// Objects: NewObjects(objects, streams, segments),
+	}
+}
+*/
```

Comment:
Because I couldn't figure out how exactly to implement this. It roughly takes these params, but I don't yet completely comprehend how to create them. It's a rough guideline how to write one.